src/backend/distributed/utils/reference_table_utils.c

```diff
@@ -426,49 +427,87 @@ CreateReferenceTableColocationId()
 
 
 /*
- * DeleteAllReferenceTablePlacementsFromNodeGroup function iterates over list of reference
- * tables and deletes all reference table placements from pg_dist_placement table
- * for given group.
+ * DeleteAllReplicatedTablePlacementsFromNodeGroup function iterates over
+ * list of reference and replicated hash distributed tables and deletes
+ * all placements from pg_dist_placement table for given group.
  */
 void
-DeleteAllReferenceTablePlacementsFromNodeGroup(int32 groupId)
+DeleteAllReplicatedTablePlacementsFromNodeGroup(int32 groupId, bool forceRemoteDelete)
 {
 	List *referenceTableList = CitusTableTypeIdList(REFERENCE_TABLE);
+	List *replicatedHashDistributedTableList =
+		ReplicatedMetadataSyncedDistributedTableList();
+
+	List *replicatedTableList =
+		list_concat(referenceTableList, replicatedHashDistributedTableList);
 
 	/* if there are no reference tables, we do not need to do anything */
-	if (list_length(referenceTableList) == 0)
+	if (list_length(replicatedTableList) == 0)
 	{
 		return;
 	}
 
 	StringInfo deletePlacementCommand = makeStringInfo();
-	Oid referenceTableId = InvalidOid;
-	foreach_oid(referenceTableId, referenceTableList)
+	Oid replicatedTableId = InvalidOid;
+	foreach_oid(replicatedTableId, replicatedTableList)
 	{
-		List *placements = GroupShardPlacementsForTableOnGroup(referenceTableId,
-															   groupId);
+		List *placements =
+			GroupShardPlacementsForTableOnGroup(replicatedTableId, groupId);
 		if (list_length(placements) == 0)
 		{
-			/* this happens if the node was previously disabled */
+			/*
+			 * This happens either the node was previously disabled or the table
+			 * doesn't have placement on this node.
+			 */
 			continue;
 		}
 
-		GroupShardPlacement *placement = (GroupShardPlacement *) linitial(placements);
+		GroupShardPlacement *placement = NULL;
+		foreach_ptr(placement, placements)
+		{
+			LockShardDistributionMetadata(placement->shardId, ExclusiveLock);
 
-		LockShardDistributionMetadata(placement->shardId, ExclusiveLock);
+			DeleteShardPlacementRow(placement->placementId);
 
-		DeleteShardPlacementRow(placement->placementId);
+			if (forceRemoteDelete)
+			{
+				resetStringInfo(deletePlacementCommand);
+				appendStringInfo(deletePlacementCommand,
+								 "DELETE FROM pg_dist_placement WHERE placementid = "
```

Comment:
would be good to always used fully qualified names in queries we send to worker nodes.